TShockAPI/Whitelist.cs

```diff
@@ -0,0 +1,291 @@
+
+	internal const string DefaultWhitelistContent = /*lang=conf*/
+		"""
+		# Localhost
+		127.0.0.1
+
+		# Uncomment to allow IPs within private ranges
+		# 10.0.0.0/8
+		# 172.16.0.0/12
+		# 192.168.0.0/16
+		""";
+
+	internal const char CommentPrefix = '#';
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="Whitelist"/> class.
+	/// Creates the whitelist file if it does not exist on disk.
+	/// </summary>
+	public Whitelist(string path)
+	{
+		_file = new(path);
+
+		if (!_file.Exists)
+		{
+			throw new FileNotFoundException("The whitelist file does not exist", _file.FullName);
+		}
+
+		ReadWhitelistFromFile();
+	}
+
+	/// <summary>
+	/// Tells if a user is on the whitelist
+	/// </summary>
+	/// <param name="host">string ip of the user</param>
+	/// <returns>true/false</returns>
+	public bool IsWhitelisted(string host)
+	{
+		if (!Enabled)
+		{
+			return true;
+		}
+
+		if (!IPAddress.TryParse(host, out IPAddress? ip))
+		{
+			throw new ArgumentException($"The provided host '{host}' is not a valid IP address.", nameof(host));
+		}
+
+		// HACK: Terraria doesn't support IPv6 yet, so we can't check for it.
+		// Remove once TShock supports IPv6.
+		if (ip.AddressFamily is AddressFamily.InterNetworkV6)
+		{
+			TShock.Log.Warn($"IPv6 address '{ip}' is not supported by Terraria. Skipping check.");
+			TShock.Log.Warn("If you somehow managed to get this message, please report it to the TShock team :");
+			TShock.Log.Warn("https://github.com/Pryaxis/TShock/issues");
+			return false;
+		}
+
+		// First check if the IP address is directly whitelisted
+		return _whitelistAddresses.Contains(ip)
+			// Otherwise is it contained within a whitelisted network?
+			|| _whitelistNetworks.Any(n => n.Contains(ip));
+	}
+
+	/// <summary>
+	/// Reloads the whitelist from the file.
+	/// </summary>
+	public void ReloadFromFile()
+	{
+		lock (_fileLock)
+		{
+			_whitelistAddresses.Clear();
+			_whitelistNetworks.Clear();
+			ReadWhitelistFromFile();
+		}
+	}
+
+	private void ReadWhitelistFromFile()
+	{
+		using StreamReader sr = _file.OpenText();
+
+		int i = 0;
+		while (!sr.EndOfStream)
+		{
+			ReadWhitelistLine(sr.ReadLine(), i);
+			i++;
+		}
+	}
+
+	private void ReadWhitelistLine(scoped ReadOnlySpan<char> content, int line)
+	{
+		// Ignore blank line or comment
+		if (content is [] or [CommentPrefix, ..] || content.IsWhiteSpace())
+		{
+			return;
+		}
+
+		// Try parse first IP range, which uses CIDR sep as discriminator.
+		if (IPNetwork.TryParse(content, out IPNetwork range))
+		{
+			_whitelistNetworks.Add(range);
+		}
+		else if (IPAddress.TryParse(content, out IPAddress? ip))
+		{
+			_whitelistAddresses.Add(ip);
+		}
+		else
+		{
+			// If we reach here, the line is not a valid IP address or network.
+			// We could throw this, but for now we just log and ignore it.
+			TShock.Log.Warn($"Invalid whitelist entry at line {line}: \"{content.ToString()}\", skipped");
```

Comment:
Adding GetString might be better
```suggestion
			TShock.Log.Warn(GetString($"Invalid whitelist entry at line {line}: \"{content.ToString()}\", skipped"));
```